datastore/kubernetes.go

```diff
@@ -624,13 +625,27 @@ func (s *DataStore) UpdateConfigMap(configMap *corev1.ConfigMap) (*corev1.Config
 // GetConfigMapRO gets ConfigMap with the given name in s.namespace
 // This function returns direct reference to the internal cache object and should not be mutated.
 // Consider using this function when you can guarantee read only access and don't want the overhead of deep copies
-func (s *DataStore) GetConfigMapRO(namespace, name string) (*corev1.ConfigMap, error) {
-	return s.cfmLister.ConfigMaps(namespace).Get(name)
+// The synchronization to lister takes a while after creating or updating a resource via Kubernetes client.
+// After creating or updating a resource, getting the resource immediately from the lister may not provide the latest result.
+func (s *DataStore) GetConfigMapRO(namespace, name string, fromLister bool) (*corev1.ConfigMap, error) {
```

Comment:
This change would make the function misleading, because the original purpose for this method is to return a cached result.

For `GetConfigMap`, it's a clone from a cached result. 

To make this simple w/o over engineering, I would suggest adding two methods to `datastore/uncached.go` and make the method names differentiated. For example, `GetConfigMapWithoutCache`. We don't need to have a RO or non-RO method in uncached, because the result is from API server directly. 

Also, we can take this time to rename methods in uncached as well to make their purpose understandable. WDYT?

Reply:
Actually, I perfer all CRUD functions have `fromCache/toCache` parameter for letting users clearly specify the target.
But I'm fine with that adding a `GetConfigMapWithoutCache` in `datastore/uncached.go`.

Reply:
This is another way to do it, but the only thing you need to take care of is no need to do an extra `DeepCopy` of the resource if it's returned from the API server directly like `GetConfigMap`, not from cache. 

Also, if we do this way, methods in uncached.go probably need to be refactored as well. This part for me looks quite confusing because it's not designed in any static way but rather a convention only.

Reply:
Yeah, handle in the future. Add `GetConfigMapWithoutCache` for the current PR.

Reply:
ref: https://github.com/longhorn/longhorn/issues/5263